### PR TITLE
Refactoring for RPC "Feature"

### PR DESCRIPTION
**Pull request by mikeolteanu**
Requires testing. What would a test look like for this? Manually setting the config to enable RPC, running the node, and running RPC commands? I see there are already RPC tests of some kind, but they seem to be more like unit tests rather than the "Is RPC running" kind of tests that this seems to require.

### CODE REVIEW

**Comment by dangershony** [comment]
Generally if all the integration tests are passing RPC is tested.
The integration tests depend on RPC working correctly.

File: Stratis.Bitcoin/RPC/RPCFeature.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    {
+        public static IFullNodeBuilder UseRPC(this IFullNodeBuilder fullNodeBuilder)
+        {
+            fullNodeBuilder.ConfigureFeature(features =>
+            {
+                features
+                .AddFeature<RPCFeature>()
+                .FeatureServices(services =>
+                    {
+                        services.AddSingleton<FullNode>();
```
Comment (dangershony) [inline Stratis.Bitcoin/RPC/RPCFeature.cs]: Full node should already be added to the DI in the builder so no need to do it here.

File: Stratis.Bitcoin/RPC/RPCFeature.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        private readonly NodeSettings nodeSettings;
+        public RPCFeature(FullNode fullNode, NodeSettings nodeSettings)
+        {
+            this.fullNode = fullNode;
+            this.nodeSettings = Guard.NotNull(nodeSettings, nameof(nodeSettings));
+        }
+
+        public override void Start()
+        {
+            if (this.nodeSettings.RPC != null)
```
Comment (dangershony) [inline Stratis.Bitcoin/RPC/RPCFeature.cs]: This is probably not the best place to check if RPC is set. 
If the start method is called on this feature its likely the implementer had the intention to enable RPC - builder.UseRPC()

I'd say at this point if RPC is null throw a NodeBuilderException

Comment (dangershony) [inline Stratis.Bitcoin/RPC/RPCFeature.cs]: As suggested by @MithrilMan log a warning if disabled in config

File: README.md [hunk 3]
```diff
@@ -6,15 +6,15 @@ https://stratisplatform.com
 Bitcoin Implementation in C#
 ----------------------------
 
-Stratis is an implementation of the Bitcoin protocol in C# on the [.NET Core](https://dotnet.github.io/) platform.  
-The daemon is a full implementation of the Stratis Bitcoin Full Node.
```
Comment (dangershony) [inline README.md]: Did you intend to remove the spaces?

File: Stratis.Bitcoin/RPC/RPCFeature.cs [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        private readonly NodeSettings nodeSettings;
+        public RPCFeature(FullNode fullNode, NodeSettings nodeSettings)
+        {
+            this.fullNode = fullNode;
+            this.nodeSettings = Guard.NotNull(nodeSettings, nameof(nodeSettings));
+        }
+
+        public override void Start()
+        {
+            if (this.nodeSettings.RPC == null)
```
Comment (dangershony) [inline Stratis.Bitcoin/RPC/RPCFeature.cs]: Don't you need to invert the condition? 
RPC != null

File: README.md (hunk 3 above)
Comment (mikeolteanu) [inline README.md]: I have a plugin on my text editor that removes trailing spaces. Generally, this is positive, except for when it adds complexity to a review...

File: Stratis.Bitcoin/RPC/RPCFeature.cs (hunk 4 above)
Comment (mikeolteanu) [inline Stratis.Bitcoin/RPC/RPCFeature.cs]: Woops, embarrassing. Good catch.

File: README.md (hunk 3 above)
Comment (dangershony) [inline README.md]: ah yeah text editors can be funny when they get too much power.

**Pull request closed without merging**